### PR TITLE
"Open page in" option for database views

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/allow-setting-default-view-type-of-a-board-particularly-on-proposals-page-8846576127534682)
image

### CODE REVIEW

File: components/[pageId]/DatabasePage/DatabasePage.tsx [hunk 1]
```diff
@@ -119,27 +120,32 @@ export function DatabasePage({ page, setPage, readOnly = false, pagePermissions
...
+  const showCard = async (cardId: string | null) => {
+    if (cardId === null) {
+      setShownCardId(null);
+      return;
+    }
+
+    const cardPage = cardId && pages ? pages[cardId] ?? (await refreshPage(cardId)) : null;
+
+    if (activeView.fields.openPageIn === 'center_peek' || cardPage?.type.includes('template')) {
+      updateURLQuery({ viewId: router.query.viewId as string, cardId });
```
Comment (mattcasey) [inline components/[pageId]/DatabasePage/DatabasePage.tsx:132]: happy to see we're trying to use the normal query for updating state

File: components/[pageId]/DatabasePage/DatabasePage.tsx [hunk 2]
```diff
@@ -119,27 +120,32 @@ export function DatabasePage({ page, setPage, readOnly = false, pagePermissions
...
-      });
-    },
-    [router.query]
-  );
+    } else if (activeView.fields.openPageIn === 'full_page' && cardPage) {
+      navigateToSpacePath(`/${cardPage.path}`);
+    }
+  };
+
+  const showView = (viewId: string) => {
```
Comment (mattcasey) [inline components/[pageId]/DatabasePage/DatabasePage.tsx:139]: why did you remove useCallback? I added this to try and avoid the database re-rendering too often. i think it was worth keeping in

File: components/common/BoardEditor/focalboard/src/components/centerPanel.tsx [hunk 3]
```diff
@@ -653,7 +664,12 @@ function CenterPanel(props: Props) {
                     readOnly={props.readOnly}
                     onCardClicked={cardClicked}
                     selectedCardIds={state.selectedCardIds}
-                    addCard={(show) => addCard('', show)}
+                    addCard={(show) =>
```
Comment (mattcasey) [inline components/common/BoardEditor/focalboard/src/components/centerPanel.tsx:667]: we should memoize all the methods passed to this component

File: components/common/BoardEditor/focalboard/src/components/viewSidebar/viewLayoutOptions.tsx [hunk 4]
```diff
@@ -22,12 +23,14 @@ interface LayoutOptionsProps {
   board?: Board;
   view: BoardView;
   intl: IntlShape;
+  hideLayoutSelectOptions?: boolean;
```
Comment (mattcasey) [inline components/common/BoardEditor/focalboard/src/components/viewSidebar/viewLayoutOptions.tsx:26]: this should be "showLayoutOptions".

File: components/common/CharmEditor/components/inlineDatabase/components/InlineDatabase.tsx [hunk 5]
```diff
@@ -91,9 +93,9 @@ export function InlineDatabase({ containerWidth, readOnly: readOnlyOverride, nod
   const views = useAppSelector((state) => selectSortedViews(state, pageId));
   const router = useRouter();
   const dispatch = useAppDispatch();
-
+  const { navigateToSpacePath } = useCharmRouter();
   const [currentViewId, setCurrentViewId] = useState<string | null>(views[0]?.id || null);
-
+  const { pages, refreshPage } = usePages();
```
Comment (mattcasey) [inline components/common/CharmEditor/components/inlineDatabase/components/InlineDatabase.tsx:98]: lets just update the serverside query in /pages/[domain]/[pageId]. we are already querying the db whenever u visit, so we could check if the pageId is a UUID and query by page id

**Review by mattcasey — CHANGES_REQUESTED** [review]
- add back useCallbacks
- instead of looking up page path, because usePages() has performance cost, i'd like to just do lookup by pageId in the server-side lookup in pages/domain/pageId.ts

File: components/[pageId]/DatabasePage/DatabasePage.tsx (hunk 2 above)
Comment (Devorein) [reply components/[pageId]/DatabasePage/DatabasePage.tsx:139]: Updated. Initially, I was trying to debug a bug that was due to a stale state, and removed the `useCallbacks`, to check whether that's the issue or not. That came out negative, but I forgot to reinstate those back.

File: components/common/BoardEditor/focalboard/src/components/centerPanel.tsx (hunk 3 above)
Comment (Devorein) [reply components/common/BoardEditor/focalboard/src/components/centerPanel.tsx:667]: Good point. Updated.

File: components/common/CharmEditor/components/inlineDatabase/components/InlineDatabase.tsx (hunk 5 above)
Comment (Devorein) [reply components/common/CharmEditor/components/inlineDatabase/components/InlineDatabase.tsx:98]: That makes sense. Updated.

**Pull request merged**